omero/developers/json-api.txt

```diff
@@ -0,0 +1,1343 @@
+within the list. In order to avoid this duplication, we can remove objects from
+within each ``omero:details`` and place them under top-level ``experimenters``
+and ``experimenterGroups`` lists.
+You can specify this with the ``?normalize=true`` query parameter.
+NB: Currently this normalizing will only apply to the top-level objects
+being listed, such as Projects, Datasets and Images. Where child objects
+are also loaded (for example Pixels within an Image), the ``omero:details``
+of these objects will not be affected by the ``?normalize=true`` parameter.
+
+
+Child counts
+^^^^^^^^^^^^
+
+For container objects such as Projects, Datasets and Screens it is
+often useful to know the number of children within them. This can be
+specified with ``?childCount=true`` parameter.
+This will add an ``omero:childCount`` value to the JSON data.
+
+
+Filtering by Owner and Group
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Most data in OMERO has an ``Owner`` and is assigned to a permissions
+``Group``. By default, queries will return data from all owners
+across all groups that are accessible to the current user.
+Use the query strings to filter by owner and/or group:
+
+::
+
+    /api/v0/m/projects/?owner=3&group=5
+
+When you are retrieving data using an object ID you will not need to
+filter by ``group`` since all the data will be in the same group.
+For example, Datasets in a specified Project will all be in the
+same group as the Project.
+
+
+Error handling
+^^^^^^^^^^^^^^
+
+Errors will result in responses with an appropriate status and may include
+JSON content with a ``message`` to provide more information:
+
+ -  **404 Not Found**: Caused by an invalid URL or when a specified
+    object can not be found in OMERO.
+
+ -  **400 Bad Request**: May be caused by invalid query parameters or
+    submitting invalid JSON content. For example,
+    ``?limit=foo`` will give a response of:
+    ::
+
+        {"message": "invalid literal for int() with base 10: 'foo'"}
+
+ -  **405 Method Not Allowed**: This will be returned if you try to use the
+    wrong http method for a url, such as ``POST`` to ``/api/v0/m/projects/``.
+    It can also be caused by trying to create or update an unsupported object,
+    such as an Image.
+
+ -  **500 Internal Server Error**: Generated from any unhandled exceptions.
+    See the ``message`` returned and check whether a ``stacktrace`` is also included.
+
+
+
+Getting started
+---------------
+
+You may find this
+`example python script <https://github.com/openmicroscopy/openmicroscopy/blob/develop/examples/Training/python/Json_Api/Login.py>`_ useful.
+It uses the python ``requests`` library to connect to the JSON api, login, query data, create and delete Projects.
+These steps are covered in more detail below.
+
+For an example how to use the API with Java, see `JSONClient.java <https://github.com/openmicroscopy/ome-documentation/blob/develop/omero/examples/java/JSONClient.java>`_.
```

Comment:
Sorry, that's what I get for looking at stuff in a rush! Yes, `:omedocs:` looks like it should work once the file is merged on develop.